test/common/http/header_utility_test.cc

```diff
@@ -34,6 +34,25 @@ class HeaderUtilityTest : public testing::Test {
   TestRequestHeaderMapImpl headers_;
 };
 
+TEST_F(HeaderUtilityTest, HasHost) {
+  const std::vector<std::pair<std::string, bool>> host_headers{
+      {"localhost", false},      // w/o port part
+      {"localhost:443", true},   // name w/ port
+      {"", false},               // empty
+      {":443", true},            // just port
+      {"192.168.1.1", false},    // ipv4
+      {"192.168.1.1:443", true}, // ipv4 w/ port
+      {"[fc00::1]:443", true},   // ipv6 w/ port
+      {"[fc00::1]", false},      // ipv6
+  };
+
+  for (const auto& host_pair : host_headers) {
+    EXPECT_EQ(HeaderUtility::getPortStart(host_pair.first) != absl::string_view::npos,
```

Comment:
Maybe check that the index is what we expect?

Reply:
I'm going to claim the use in actual port removal validates that, sound reasonable?
(working on the e2e test either way)

Reply:
Yep, sounds good!